consumer/consumer.go

```diff
@@ -0,0 +1,88 @@
+package consumer
+
+import (
+    "log"
+    "github.com/satori/go.uuid"
+    "github.com/streadway/amqp"
+)
+
+type Consumer struct {
+    uri string
+    exchangeName string
+    conn *amqp.Connection
+    ch *amqp.Channel
+    Deliveries <-chan amqp.Delivery
+    Err chan error
+}
+
+func New(uri string, exchangeName string) *Consumer {
+    c := &Consumer{}
+    c.uri = uri
+    c.exchangeName = exchangeName
+    c.Err = make(chan error)
+
+    return c
+}
+
+func (c *Consumer) Start() () {
+    var err error
+    c.conn, c.ch, err = c.openChannel()
```

Comment:
Should I return err in the c.Err channel after each possible error ?
`    c.Err <- err`